components/KiboCollectedProduct.vue

```diff
@@ -178,13 +211,44 @@ export default defineComponent({
       await removeCartItem(props.cartItemId)
     }
 
+    // Get Fullfillment Options
+    const handleFulfillmentOption = async (
+      selectedFulfillmentValue?: string,
+      shouldOpenModal?: boolean
+    ) => {
+      const itemToBeUpdated = { ...cart.value.items.find((item) => item.id === props.cartItemId) }
+      if (selectedFulfillmentValue === "Ship") {
+        itemToBeUpdated.fulfillmentMethod = selectedFulfillmentValue
+        itemToBeUpdated.purchaseLocation = null
+        await updateCartItem(props.cartItemId, itemToBeUpdated)
+      }
+
+      if (shouldOpenModal) {
+        modal.show({
+          component: StoreLocatorModal,
+          props: {
+            title: context?.root?.$t("Select Store"),
+            handleSetStore: async (selectedStore: string) => {
+              itemToBeUpdated.fulfillmentMethod = "Pickup"
+              itemToBeUpdated.fulfillmentLocationCode = selectedStore
+              itemToBeUpdated.purchaseLocation = selectedStore
+              await updateCartItem(props.cartItemId, itemToBeUpdated)
+            },
```

Comment:
If itemToBeUpdated is null or undefined then code will break to avoid it can we gard it using either if statement or Optional chaining (?.) operator?